File: 00-solution/do/go/main.go
package main

import (
	"github.com/pulumi/pulumi-digitalocean/sdk/v4/go/digitalocean"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		clusterRegion := "fra1"
		nodePoolName := "default"
		nodeCount := 1
		version := "1.31.1-do.5"
		doCluster, err := digitalocean.NewKubernetesCluster(ctx, "do-cluster", &digitalocean.KubernetesClusterArgs{
			Region:                        pulumi.String(clusterRegion),
			Version:                       pulumi.String(version),
			DestroyAllAssociatedResources: pulumi.Bool(true),
			NodePool: &digitalocean.KubernetesClusterNodePoolArgs{
				Name:      pulumi.String(nodePoolName),
				Size:      pulumi.String("s-2vcpu-2gb"),
				NodeCount: pulumi.IntPtr(nodeCount),
			},
		})
		if err != nil {
			return err
		}
		ctx.Export("name", doCluster.Name)
		ctx.Export("kubeconfig", doCluster.KubeConfigs.ApplyT(func(kubeConfigs []digitalocean.KubernetesClusterKubeConfig) (*string, error) {
			return kubeConfigs[0].RawConfig, nil
		}).(pulumi.StringPtrOutput))
		return nil
	})
}

File: 00-solution/do/python/__main__.py
import pulumi
import pulumi_digitalocean as digitalocean

cluster_region = "fra1"
node_pool_name = "default"
node_count = 1
version = "1.31.1-do.5"
do_cluster = digitalocean.KubernetesCluster("do-cluster",
    region=cluster_region,
    version=version,
    destroy_all_associated_resources=True,
    node_pool={
        "name": node_pool_name,
        "size": "s-2vcpu-2gb",
        "node_count": node_count,
    })
pulumi.export("name", do_cluster.name)
pulumi.export("kubeconfig", do_cluster.kube_configs[0].raw_config)
